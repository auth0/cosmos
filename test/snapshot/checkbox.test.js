import React from 'react'
import renderer from 'react-test-renderer'
import { Checkbox } from '@auth0/cosmos'
import 'jest-styled-components'

describe('Checkbox', () => {
  describe('renders with automation attribute', () => {
    it('single checkbox', () => {
      const tree = renderer.create(<Checkbox name="cosmos" />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('group of checkboxs', () => {
      const tree = renderer.create(<Checkbox.Group name="cosmos" />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
