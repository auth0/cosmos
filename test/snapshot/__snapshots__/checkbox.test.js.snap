// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders with automation attribute group of checkboxs 1`] = `
.c0 .c1 {
  display: table;
}

.c0 .c1:last-child {
  margin: 0;
}

<div
  className="c0"
  data-cosmos-key="checkbox.group"
  name="cosmos"
/>
`;

exports[`Checkbox renders with automation attribute single checkbox 1`] = `
.c0 {
  position: relative;
  cursor: pointer;
  padding-left: 24px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
}

.c0 input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
}

.c0 .c1 {
  position: absolute;
  top: 50%;
  left: 0;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  height: 16px;
  width: 16px;
  background-color: #FFF;
  border: 1px solid #BEBEBE;
  box-shadow: inset 0 1px 2px 0 rgba(0,0,0,0.20);
  border-radius: 2px;
}

.c0:hover input ~ .c1 {
  box-shadow: inset 0 1px 4px 0 rgba(0,0,0,0.20);
}

.c0 input:checked ~ .c1 {
  background-color: #3B99FC;
  border: 1px solid #2C90FC;
}

.c0 .c1:after {
  content: '';
  position: absolute;
  display: none;
}

.c0 input:checked ~ .c1:after {
  display: block;
}

.c0 .c1:after {
  width: 4px;
  height: 8px;
  border: solid #FFF;
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg) translate(-50%,-50%);
  -ms-transform: rotate(45deg) translate(-50%,-50%);
  transform: rotate(45deg) translate(-50%,-50%);
  left: 20%;
  top: 50%;
}

<label
  className="c0"
  data-cosmos-key="checkbox"
>
  <input
    name="cosmos"
    readOnly={true}
    type="checkbox"
  />
  <span
    className="c1 "
  />
  <span
    className=""
  />
</label>
`;
